✨(courses) allow synchronizing course runs via synchronization hooks

sampaccoud opened the pull request:
## Purpose

The courses in our LMS are indexed on a number of external catalogs that must be kept up to date. This can be done by emitting calls to external web hooks.

## Proposal

Use the existing asynchronous task called when a course is published to make calls to external web hooks declared via a new `COURSE_HOOKS" setting.

sampaccoud committed:
✨(courses) allow synchronizing course runs via synchronization hooks

Courses in our LMS are indexed on a number of external catalogs that must be
kept up to date. This can be done by emitting calls to external web hooks.

lunika commented on courses/tasks.py:
```diff
@@ -1,15 +1,64 @@
+    A task to call when a course is updated in OpenEdX.
+    It calls:
+    - the "update_courses" management command to update fun-apps' search index.
+    - each course run hook url defined in the `COURSE_HOOKS` settig with scheduling information
```
```suggestion
    - each course run hook url defined in the `COURSE_HOOKS` setting with scheduling information
```

sampaccoud replied:
:+1:

lunika commented on courses/tasks.py:
```diff
@@ -1,15 +1,64 @@
+            requests.post(
+                hook["url"],
+                data=data,
+                headers={'Authorization': authorization},
```
This is directly linked to this comment https://github.com/openfun/richie/pull/1226/files#r546885776 once we decided what we do we should report (or not) the modification

sampaccoud replied:
done

lunika commented on courses/tasks.py:
```diff
@@ -1,15 +1,64 @@
+    for hook in hooks:
+        authorization = hmac.new(
+            hook["secret"].encode("utf-8"),
+            msg=json.dumps(data).encode("utf-8"),
```
does the `json.dumps` can be computed outside of the loop ? Doing this, does it avoid unnecessary computing ? Or python is smart enough to cache this instruction ?

sampaccoud replied:
good point! :pray:

Commit:
🔖(minor) bump version to 5.7.0

Added

- Allow synchronizing course runs via an external synchronization hook

Fixed

- Fix "update_courses" command used in cron job for mass updates

jbpenrath reviewed (approved):
👏
